docs/posts/lms-for-data/index.qmd

```diff
@@ -0,0 +1,690 @@
+---
+title: "Using language models for data"
+author: "Cody Peterson"
+date: "2024-02-05"
+image: thumbnail.png
+categories:
+    - blog
+    - llms
+    - duckdb
+---
+
+## Overview
+
+This post will give an overview of how (large) language models (LMs) fit into
+data engineering, data analyst, and data science workflows.
+
+## Use cases
+
+There are three main use cases for language models for data practitioners:
+
+1. Synthetic data generation
+2. Natural language processing
+3. Writing analytic code
+
+We'll describe each and then demonstrate them with code.
+
+## Setup
+
+We'll use [Marvin](https://askmarvin.ai), the AI engineering toolkit, alongside
+[Ibis](https://ibis-project.org), the data engineering toolkit, to demonstrate
+the capabilities of language models for data using the default DuckDB backend.
+
+:::{.callout-tip}
+We'll use a cloud service provider (OpenAI) to demonstrate these capabilities.
+In a follow-up post, we'll explore using local open source language models to
+achieve the same results.
+:::
+
+With Marvin and Ibis, you can replicate the workflows below using other AI
+service providers, local language models, and over 20+ data backends!
+
+You'll need to install Marvin and Ibis to follow along:
+
+```{.bash}
+pip install 'ibis-framework[duckdb,examples]' marvin
+```
+
+Then import them and turn on Ibis interactive mode:
+
+```{python}
+import ibis  # <1>
+import marvin  # <2>
+
+from pydantic import BaseModel, Field  # <3>
+
+ibis.options.interactive = True  # <4>
+ibis.options.repr.interactive.max_rows = 6  # <5>
+```
+
+```{python}
+# | code-fold: true
+# | echo: false
+_ = ibis.get_backend().raw_sql("set enable_progress_bar = false")
+```
+
+1. Import Ibis, the data engineering toolkit
+2. Import Marvin, the AI engineering toolkit
```

Comment:
why marvin? calling marvin `the` AI engineering toolkit is a bit too strong. are they our partner?

Reply:
Just a comment on the label, it seems okay to call it the AI engineering toolkit given that it's on their website (https://askmarvin.ai)

![image](https://github.com/ibis-project/ibis/assets/417981/1649f6ab-ca05-404f-ae60-c3b226853582)

Reply:
yeah it's their tagline. perhaps ambitious but the only useful abstraction I've found in Python for working with LLMs, handling the logit bias trick for classification and schematization for other purposes

makes it easy and Pythonic to do just about anything you'd want to

Reply:
and they are not a partner

Reply:
> why marvin? calling marvin `the` AI engineering toolkit is a bit too strong. are they our partner?

Similar thought, I believe our team already invested lots of time on the toolkit, I am not very sure how popular marvin is.